llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -92,6 +116,33 @@ def __init__(self, config: MPTConfig):
                 **config.to_dict(),
             ) for _ in range(config.n_layers)
         ])
+        if config.tensor_parallel_mlp and is_torch_2_or_higher():
+            if enable_2d_with_fsdp():
+                device_type = 'cuda' if get_device(
+                    None).name == 'gpu' else 'cpu'
+                world_size = dist.get_world_size()
+                node_count = world_size // dist.get_local_world_size()
```

Comment:
could you add a comment here that this assumes a constant local world size across nodes (or only one node is smaller)? We have a TODO to at some point make sure everything (composer, foundry, streaming) work with non-constant local world size.